test/Cimpress.Extensions.Http.UnitTests/MessageHandlers/MeasurementHandler_when_logging_results.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Cimpress.Extensions.Http.MessageHandlers;
+using FluentAssertions;
+using Xunit;
+
+namespace Cimpress.Extensions.Http.UnitTests.MessageHandlers
+{
+    public class MeasurementHandler_when_logging_results
+    {
+        private class TestHandler : DelegatingHandler
+        {
+            private readonly TimeSpan executionTime;
+            private readonly HttpStatusCode statusCode;
+            private readonly bool throwException;
+
+            public TestHandler(TimeSpan executionTime, HttpStatusCode statusCode, bool throwException)
+            {
+                this.executionTime = executionTime;
+                this.statusCode = statusCode;
+                this.throwException = throwException;
+            }
+
+            protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                await Task.Delay(executionTime, cancellationToken);
+                if (throwException)
+                {
+                    throw new Exception("unit test");
+                }
+                return new HttpResponseMessage(statusCode);
+            }
+        }
+
+        public static IEnumerable<object[]> LogInput
+        {
+            get
+            {
+                yield return new object[] {new Uri("http://unit.test"), HttpStatusCode.OK, false, true, false, false};
+                yield return new object[] {new Uri("http://unit.test/foo/bar"), HttpStatusCode.BadRequest, false, true, false, false};
+                yield return new object[] {new Uri("http://unit.test/xyz.html"), HttpStatusCode.OK, true, true, true, false};
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(LogInput))]
+        public async Task Only_logs_when_completed(Uri uri, HttpStatusCode statusCode, bool throwException, bool isCompleted, bool isFaulted, bool isCanceled)
```

Comment:
I think we also need the test, awaited task doesn't wait for log func.  Tested by passing in a logging func which will will block forever...